src/Nethermind/Nethermind.Monitoring/Metrics/MetricsController.cs

```diff
@@ -24,64 +26,158 @@ namespace Nethermind.Monitoring.Metrics
     public partial class MetricsController : IMetricsController
     {
         private readonly int _intervalSeconds;
-        private Timer _timer;
-        private readonly Dictionary<Type, (MemberInfo, string, Func<double>)[]> _membersCache = new();
-        private readonly Dictionary<Type, DictionaryMetricInfo[]> _dictionaryCache = new();
+        private Timer _timer = null!;
+        private static bool _staticLabelsInitialized = false;
+
+        private readonly Dictionary<Type, IMetricUpdater[]> _metricUpdaters = new();
         private readonly HashSet<Type> _metricTypes = new();
 
-        public readonly Dictionary<string, Gauge> _gauges = new();
+        // Largely for testing reason
+        public readonly Dictionary<string, IMetricUpdater> _individualUpdater = new();
```

Comment:
Make it private and add public property of type `IReadOnlyDictionary`.
Or at least internal and use `InternalsVisibleTo` attribute.